File: src/main/java/net/pterodactylus/sone/database/FriendProvider.java
package net.pterodactylus.sone.database;

import java.util.Collection;

import net.pterodactylus.sone.data.Sone;

/**
 * Provides information about {@link Sone#getFriends() friends} of a {@link Sone}.
 *
 * @author <a href="mailto:bombe@pterodactylus.net">David ‘Bombe’ Roden</a>
 */
public interface FriendProvider {

	Collection<String> getFriends(Sone localSone);
	boolean isFriend(Sone localSone, String friendSoneId);

}

File: src/main/kotlin/net/pterodactylus/sone/database/FriendProvider.kt
package net.pterodactylus.sone.database

import net.pterodactylus.sone.data.Sone

/**
 * Provides information about [friends][Sone.getFriends] of a [Sone].
 */
interface FriendProvider {

	fun getFriends(localSone: Sone): Collection<String>
	fun isFriend(localSone: Sone, friendSoneId: String): Boolean

}
